src/org/labkey/test/tests/FolderTest.java

```diff
@@ -255,6 +272,119 @@ public void folderTemplateTest()
+        clickButton("Delete", WAIT_FOR_PAGE);
+
+        //now delete multiple workbooks:
+        _containerHelper.createSubfolder(getProjectName(), "WorkbookParent");
+        goToProjectHome(getProjectName() + "/WorkbookParent");
+        PortalHelper ph = new PortalHelper(this);
+        ph.addBodyWebPart("Workbooks");
+        DataRegionTable dr = DataRegionTable.findDataRegionWithinWebpart(this, "Workbooks");
+
+        WorkbookHelper workbookHelper = new WorkbookHelper(this);
+        workbookHelper.createWorkbook(getProjectName() + "/WorkbookParent", "WB1", "Description", WorkbookHelper.WorkbookFolderType.DEFAULT_WORKBOOK);
+        int wb = workbookHelper.createWorkbook(getProjectName() + "/WorkbookParent", "WB2", "Description", WorkbookHelper.WorkbookFolderType.DEFAULT_WORKBOOK);
+        createListWithData("WorkbookParent/" + wb);
+
+        workbookHelper.createWorkbook(getProjectName() + "/WorkbookParent", "WB3", "Description", WorkbookHelper.WorkbookFolderType.DEFAULT_WORKBOOK);
+        goToProjectHome(getProjectName() + "/WorkbookParent");
+        dr = DataRegionTable.findDataRegionWithinWebpart(this, "Workbooks");
+
+        assertEquals("Incorrect row count", 3, dr.getDataRowCount());
+
+        dr.checkCheckbox(1);
+        dr.checkCheckbox(2);
+        dr.clickHeaderButton("Delete");
+        assertTrue(acceptAlert().contains("Are you sure you want to delete the selected rows?"));
+        waitForElement(Locator.tagWithText("td", "You are about to delete the following folders:"));
+        waitForElement(Locator.tagContainingText("li", getProjectName() + "/WorkbookParent/2, containing the following objects"));
+        waitForElement(Locator.tagWithText("li", "1 lists"));
+        waitForElement(Locator.tagContainingText("li", getProjectName() + "/WorkbookParent/3 and all the objects it contains"));
+        clickButton("Delete", WAIT_FOR_PAGE);
+
+        //make sure we cannot delete using the client API
+        goToProjectHome(getProjectName() + "/WorkbookParent");
+        try
+        {
+            Connection cn = new Connection(WebTestHelper.getBaseURL(), PasswordUtil.getUsername(), PasswordUtil.getPassword());
+            SelectRowsCommand sr = new SelectRowsCommand("core", "workbooks");
+            sr.setColumns(Arrays.asList("EntityId"));
+            SelectRowsResponse srr = sr.execute(cn, getProjectName() + "/WorkbookParent");
+            Object entityId = srr.getRows().get(0).get("EntityId");
+
+            DeleteRowsCommand drc = new DeleteRowsCommand("core", "workbooks");
+            Map<String, Object> row1 = new HashMap<>();
+            row1.put("EntityId", entityId);
+            drc.addRow(row1);
+
+            SaveRowsResponse resp = drc.execute(new Connection(WebTestHelper.getBaseURL(), PasswordUtil.getUsername(), PasswordUtil.getPassword()), getContainerId());
+
+            throw new Exception("That command should have failed");
+        }
+        catch (IOException | CommandException e)
```

Comment:
Don't bother catching `IOException`. If that gets thrown then something more serious has gone wrong.
```suggestion
        catch (CommandException e)
```

Reply:
ok